src/Aquifer.Common/Utilities/VersificationUtilities.cs

```diff
@@ -1,19 +1,258 @@
+        var targetBibleVerseIdByBaseVerseIdWithOptionalPartMap =
+            await versificationService.GetBibleVerseIdByBaseVerseIdWithOptionalPartMapAsync(targetBibleId, ct);
+        var targetBibleExcludedVerseIds = await versificationService.GetExcludedVerseIdsAsync(targetBibleId, ct);
+
+        return ConvertVersificationCore(
+            sourceVerseId,
+            baseVerseIdWithOptionalPartBySourceBibleVerseIdMap,
+            targetBibleVerseIdByBaseVerseIdWithOptionalPartMap,
+            targetBibleExcludedVerseIds);
+    }
+
+    /// <summary>
+    /// This method assumes that the <paramref name="sourceVerseId"/> is valid for the given maps and exclusions.
+    /// </summary>
+    /// <returns>The converted verse ID or <c>null</c> if conversion is not possible.</returns>
+    private static int? ConvertVersificationCore(
+        int sourceVerseId,
+        ReadOnlyDictionary<int, string> baseVerseIdWithOptionalPartBySourceBibleVerseIdMap,
+        ReadOnlyDictionary<string, int> targetBibleVerseIdByBaseVerseIdWithOptionalPartMap,
+        ReadOnlySet<int> targetBibleExcludedVerseIds)
+    {
         // The dictionary only contains mappings where the key is different from the value.
         // If the key verse ID is not present in the mapping (and is not in the exclusions list) then the value matches the key.
-        var baseVerseId = baseVerseIdBySourceBibleVerseIdMapping.GetValueOrDefault(sourceBibleVerseId, sourceBibleVerseId);
+        var baseVerseIdWithOptionalPart = baseVerseIdWithOptionalPartBySourceBibleVerseIdMap.GetValueOrDefault(sourceVerseId)
+            ?? sourceVerseId.ToString();
+
+        // Base Verse Data includes a Verse ID with an optional verse part at the end (such as 'a', 'b', etc.).
+        var baseVerseId = baseVerseIdWithOptionalPart[..10];
+
+        // Rules when a part is present on base verse ID:
+        // 1. Use the base verse ID with part if present in the target inverse mapping (the direct map).
+        // 2. If not found, then try to map using the base verse ID without a part (in case the target mapping doesn't use parts).
+        // 3. Default to the base verse ID without a part (no explicit mapping).
+        // Rules when a part is not present on base verse ID:
+        // 1. Use the base verse ID without a part if present in the target inverse mapping (the direct map).
+        // 2. If not found, then try to map using the base verse ID with part 'a' (in case the target mapping uses parts).
+        // 3. Default to the base verse ID without a part (no explicit mapping).
+        var targetVerseId = targetBibleVerseIdByBaseVerseIdWithOptionalPartMap.GetValueOrNull(baseVerseIdWithOptionalPart)
+               ?? targetBibleVerseIdByBaseVerseIdWithOptionalPartMap.GetValueOrNull(
+                   baseVerseIdWithOptionalPart.Length > 10
+                       ? baseVerseId
+                       : $"{baseVerseId}a")
```

Comment:
Interestingly, there are multiple mappings where the first `baseVerseIdPart` is `b`.  In those cases this lookup will miss that first mapping (because we append `a`) and the lookup will then miss any explicit mapping and use the default of `value = key`.  Looking at the data where `b` is the only mapping this logic still seems like the best move (and I added integration tests for a couple of them).